pandas/core/dtypes/cast.py

```diff
@@ -1145,12 +1145,16 @@ def convert_dtypes(
                 and isna(input_array).all()
             ):
                 import pyarrow as pa
-
+                
                 pa_type = pa.null()
             else:
                 pa_type = to_pyarrow_type(base_dtype)
             if pa_type is not None:
-                inferred_dtype = ArrowDtype(pa_type)
+                if isna(input_array).all() and hasattr(input_array, 'categories'):
+                    inferred_dtype = input_array.dtype
+                else:
+                    inferred_dtype = ArrowDtype(pa_type)
```

Comment:
Instead of this, can you add the check `isinstance(input_array.dtype, CategoricalDtype)` to the if statement that starts on L1142. Then in the case to categorical dtypes, `to_pyarrow_type` returns `None` on L1151 and we do not modify `inferred_dtype`.

Reply:
good point!